### DISCUSSION TITLE
V3.08 released

### DISCUSSION THREAD

**Discussion by GerryDazoo** [GerryDazoo/Slinger#136] [Announcements]
Please have a look at the release notes. There's some good stuff in there. This version should be backwards compatible with your current configuration but there's some new features to help support people with multiple slingboxes. It supports automatically changing to your favourite channel when you start up a stream so in many cases you won't need to start a remote to change channels.
So if you have a working setup and don't want to mess with this you don't need to upgrade. I'll support both versions for now.

[V3.08_release_notes.pdf](https://github.com/GerryDazoo/Slinger/files/9795067/V3.08_release_notes.pdf)

**Comment by 89mclarenstang** [comment]
Thanks for the update!  Will this also be able to change the channel for us when using it with xteve and Plex?  The part that rbf121 had worked on?

**Comment by GerryDazoo** [comment]
Yes. Please read the release notes.

**Comment by floormanager1** [comment]
Great work Gerry!  I'm up and running (remembering to remove my finderIDs ;-)  Yet to try the multiple Slingboxes on the server but if I can work it out, it will be a great feature.

**Reply by GerryDazoo** [reply]
I may have gone a little bit overboard on the finderid thing. :-)
The new code tries a few times and then gives up. Before it would try forever every 10 seconds.

**Comment by DoogieHowitzerWMD** [comment]
Would I be able to use the new config file with the custom URL base, but have one such config file for each slingbox?

Or could I add the URL base to the old config files and just run the new server.py file?

Basically, I just want to keep each slingbox on the separate port numbers already assigned, but having the ability to change the URL base would be a nice feature.

**Reply by GerryDazoo** [reply]
You should just be able to add the URLBase to each of your config.ini files

**Reply by DoogieHowitzerWMD** [reply]
Cool, thanks. I'll give that a shot.

**Comment by mdehrlich** [comment]
Gerry,  Very nice redo on the design.  I apologize for enabling the cloud registration.  I thought the finderid was always required to make the slinger work.  I've commented them out in the new config.

**Comment by GR0DZ1LLA** [comment]
Damn Gerry. Good work. Putting out updates faster than I can get time to test all these changes. Appreciate it all. Can’t wait to try all the things.

**Comment by BobC1954** [comment]
Can confirm working with Echostar HDS-600rs.

**Comment by tom76017** [comment]
Looks good.  I had to comment out line 779 (#  rccode = remotes[box_name][2]) since I am not enabling remotes.  VCL would not paint.

Thanks for the great effort!

**Reply by GerryDazoo** [reply]
I'll take a look. Definitely a bug. I need to fire my system test team!

**Reply by tom76017** [reply]
Thx. 
I've loaded 308c and it's working fine. I've started looking at resolution. VLC says I'm getting 720. My TiVo Mini and M1 specs say 1080 is supported. Is there any more documentation on this in the config.ini:
_

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12
_

**Reply by GerryDazoo** [reply]
You can thank the Sling marketing guys for this. What they REALLY meant to say it supports 1080 input not output.
I do know the 500 will generate a 1080i output at 30 Hz. But my M1 will eat 1080i but won't stream at that resolution (hardware restrictions I suspect). I think somewhere in the sling FAQs this is covered.
Have a close look at the stream info. What I get for 720 when I feed the box 1080i is actually 1280x540. The box seems to be just throwing away every other scan line.

**Reply by mdehrlich** [reply]
Note that the resolutions go from 1 to 16.  1920X1280 does work if you use resolution 16.
I cannot get 720.  Sounds like you also don't get it.

**Reply by GerryDazoo** [reply]
I can get 1280x720 but you have to feed 1280x720 into the slingbox not 1920x1280

**Reply by GerryDazoo** [reply]
https://www.slingbox.com/help/kb/KB-2000580
https://www.slingbox.com/help/KB/KB-2000464

**Comment by rblancarte** [comment]
Just installed 3.08 to test out rbf121's Plex setup which I was able to get working.

However, this new hitch raised a small issue for me:
1. I am using a Tivo Mini which I let sleep most of the time it isn't in use. I need to send 2 LIVE TV button presses to the TV to make it activate the tuner. Is there a way to make that my start channel? IE, not just `StartChannel=155`
2. When I use the URL channel changing `?channel=155` the key presses aren't fast enough and they don't all get accepted by the Tivo (or they are mucked up and I actually end up on a single digit channel, IE I enter 513, I end up on channel 3). Any thoughts?

**Reply by GerryDazoo** [reply]
Not sure what you mean by new hitch? 
1) Not without me making some code changes.
2) Many people are using this with the Tivo and haven't reported any issues. Does it work correctly when you use the web Remote to change the channel? 

Can you move this to a new separate discussion discussion.

**Reply by rblancarte** [reply]
hitch - sorry, a remnant from an edit. That was supposed to say new release.

1. I might dig through the code on this, I could see a good use for allowing direct button presses in some instances.
2. I will start a new topic on this. - https://github.com/GerryDazoo/Slinger/discussions/161

**Comment by BobC1954** [comment]
Gerry. Can the server be made to re-try rather than giving up if it doesn't find a slingbox? (Or maybe it does but I'm not waiting long enough). Windows version of the server. All my Slingboxes are on IP controlled power switches and I only turn on when I want to watch - to save power.

**Reply by GerryDazoo** [reply]
It could. How are you running the slingbox_server, manually? But in the meantime, you could always write a little script to ping the boxes to make sure they are alive before running the server.

**Reply by BobC1954** [reply]
Currently running it manually, but also tried .bat and will use that method when I've got everything working and then copied onto my little Windoze server.

**Reply by GerryDazoo** [reply]
Yes, a few lines of code in the .bat should do the trick.

**Reply by BobC1954** [reply]
Gerry. In regard to remote not working with ProHD. I don't know if this is any help, but the Keycodes line in the server shows Keycodes = [] for ProHD, yet shows Keycodes = [1, 4, 5, 9 etc etc...] for the HDS-600rs and that works.

**Reply by GerryDazoo** [reply]
I'm working on it. Getting very close. Standby.

**Comment by tom76017** [comment]
I use a separate app after activating my Mini's Network Remote Control.  It's faster since not buffered.

See https://github.com/GerryDazoo/Slinger/discussions/categories/show-and-tell

![image](https://user-images.githubusercontent.com/26171888/197201245-46057518-9d41-4452-875a-7eeaa88e383d.png)

**Comment by theweatherguy** [comment]
OK. Found the Announcements Conversation.

**Comment by almoney** [comment]
Thank you for all the hard work!

I wanted to ask if you have thought about adding the ability to configure the slingboxes after a hard reset?  I am concerned that since I let my boxes switch to uPNP port forwarding that the ports will change after router or slingbox restart.  I may do some quick testing.  If the setup on Sling.com still allows hard port assignments I may do that for now, but that will only work until the dish shutdown of the cloud service.

**Reply by DoogieHowitzerWMD** [reply]
> Thank you for all the hard work!
> 
> I wanted to ask if you have thought about adding the ability to configure the slingboxes after a hard reset? I am concerned that since I let my boxes switch to uPNP port forwarding that the ports will change after router or slingbox restart. I may do some quick testing. If the setup on Sling.com still allows hard port assignments I may do that for now, but that will only work until the dish shutdown of the cloud service.

Hmm... I'm wondering what a hard reset would do in terms of reverting the slingbox back to the factory-preset firmware. Once Sling's servers are gone, the factory preset firmware is what you'd be stuck with on the box. I suppose it could still be capable of connecting to our self-hosted server, but I wonder if that is verifiable.

**Reply by BobC1954** [reply]
My suspicion is that older Slingboxes (Pro, ProHD etc) "may" still be able to be set up, using early versions of Slingplayer - up to v2.0.3 approx. Later versions and later Slingboxes require access to Sling's servers - I think. Further more, versions of Slingplayer later than about 2.0 require flash in order to provide the tuning and setup routines, but version 1.xx will work standalone. I guess we'll have to wait and see what happens in November. That's how I see things at the moment.

**Comment by theweatherguy** [comment]
Hi Gerry,  I'm having some issues with 3.08c on my Mac.   Several errors.   I have two 350 slingboxes.  One called Den and one called Carol.  Below is Terminal Output at startup showing the errors.   Below that is my unified_config.ini file.  Maybe I'm not doing this right?   I'm a little confused about whether naming the individual boxes like [Den] and [Carol] in the config code has screwed things up. 

Anyhow, I went back to version 3.07 for now.  BTW, with the reversion back to 3.07, I found I needed to explicitly specify the video source as 1 (where before I had left as commented out)   I'll have to tackle this over the weekend.



**Here's my terminal output:**

 slingbox$ python3 slingbox_server.py unified_config.ini
Version : 3.08c Running on macOS-12.6-x86_64-i386-64bit pid= 1086
Using config file unified_config.ini
Connection Manager Running on port 8080 with 3 max streams using URL slingbox.
BOXES [('sb1', '350'), ('sb2', 'Carol')]
Warning TCP buffering might not be sufficent for reliable streaming
Warning TCP buffering might not be sufficent for reliable streaming
Streamer Running:  3 unified_config.ini Carol Carol 8080 5242880
Streamer Running:  3 unified_config.ini 350 350 8080 5242880
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
Warning TCP buffering might not be sufficent for reliable streaming
Checking for slingbox at ('192.168.2.132', 5201)
    self.run()
starting up on  port 8080 5242880
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
('192.168.2.132', 5201) OK
Carol Using slingbox at  ('192.168.2.132', 5201)
    self._target(*self._args, **self._kwargs)
  File "/Users/slingbox/slingbox_server.py", line 579, in streamer
    slinginfo = cp[section_name]
Streamer:  Carol Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 964, in __getitem__
    raise KeyError(key)
KeyError: '350'
Building page for 350
Traceback (most recent call last):
  File "/Users/slingbox/slingbox_server.py", line 1337, in <module>
    remotes[box[1]] = BuildPage(cp, box[1])
  File "/Users/slingbox/slingbox_server.py", line 1225, in BuildPage
    remoteinfo = cp[section_name]
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 964, in __getitem__
    raise KeyError(key)
KeyError: '350'




**Here's my unified_config file.**  (obscured my passwords for privacy)

; This a sample of a unified config.ini for three slingboxes

[SLINGBOXES]
sb1=Den
sb2=Carol
;sb3=M1

[Den]
sbtype="350/500/M1"
password=9TsYINoTREASLPASSWORDoerb
;finderid=nogood4CF70531865D73650A16A0A536
ipaddress=192.168.2.146
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=245
;RemoteLock=yes
Remote=350_remote.txt

[Carol]
sbtype="350/500/M1"
password=sisZ1NOTREAOLPASSORDTer
ipaddress=192.168.2.132
;finderid=BAD95DAED609364C8204F2BC5FBDC363
port=5201
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
;RemoteLock=yes
;StartChannel=155
Remote=Carol_remote.txt

;[M1]
;sbtype="350/500/M1"
;password=SOmePassword
;finderid=E706A7405882C03A076502246DAD632F
;ipaddress=192.168.117.129
;port=5301
;Resolution=12
;FrameRate=30
;VideoBandwidth=2000
;VideoSmoothness=63
;VideoSource=1
;Remote=m1_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=8080
maxremotestreams=3
URLbase=slingbox
enableremote=yes

**Reply by GerryDazoo** [reply]
You're not using the unified config file you think you are;
BOXES [('sb1', '350'), ('sb2', 'Carol')]
Should say sb1='Den' if you were using the same file that you posted.

That said, there is a small issue. The Slingbox name should not be all digits. It confuses the code when it might also be trying to find a SB based on a port number.

PS. You don't need to comment out the M1 section. It won't be used if it's not listed in the [SLINGBOXES] section.

**Reply by theweatherguy** [reply]
Thanks.  Yes, I now see I made the error with sb1= Den.  And I’ll change the names to be not all numeric.  Will try again in a couple of days.

**Reply by theweatherguy** [reply]
Gerry,
I got 3.08c working just super on my Mac.   Just needed to change the names as you had pointed out.  I'll post a fresh Mac Setup Step by Step for the new version over the next few days.  

Thank you so much for creating this!  I can't tell you how thrilled my wife and I are about not losing our slingboxes after Nov 9th!

**Comment by mdehrlich** [comment]
I also noted your slingboxes. have the same port numbers?    May make it more difficult if you do internet viewing

**Reply by theweatherguy** [reply]
It hasn’t been a problem with v 3.07.  I'm using a Wireguard VPN server setup in my home and I’m connecting without any issues.
